### ISSUE TITLE
sorted() for datapath means that inside a batch, data are grouped and unchanged?

### ISSUE THREAD

**Issue by Sydhm** [junyanz/pytorch-CycleGAN-and-pix2pix#1263]
Hi there, thank you for such great work in this repo. However, while trying to create a custom dataset, I realize that in Cyclegan, even when using unpaired data, a single batch is unchanged during every epoch of the training stage. Namely data_A1 is paired with data_B1 for every epoch (because of the use of sorted() function), shuffling the dataloader however, will only change the order the pair is accessed in the training epoch. Is that intended? Does that mean using a zip of 2 dataloaders for A and B with shuffling might have an impact on the performance of cyclgegan?

Thank ahead for any help!
